tests/integration/s2n_handshake_test_s_client.py

```diff
@@ -318,6 +323,10 @@ def try_handshake(endpoint, port, cipher, ssl_version, server_name=None, strict_
     s_client_out += read_process_output_until(s_client, openssl_connect_marker)
     s2nd_out += read_process_output_until(s2nd, openssl_connect_marker)
 
+    # Fails the test if any of the process returned abnormally
+    if s_client.returncode or s2nd.returncode:
+        return -1
```

Comment:
1. Is this the right place for this, or wouldn't it need to come after all the possible read_process_output_untils?
2. Don't you need to clean up any processes that didn't return? Otherwise we get port binding issues

Reply:
looks like i might have to move things around and do this check after each process wait group :( if we dont handle exiting processes - other parts of this file would also crash.